File: Xenon-Vis_Studio/Xenon/Missile.ts
﻿import gsCControls = require("Controls");
import CScene = require("Scene");
import CBullet = require("Bullet");
import CActor = require("Actor");
import enums = require("Enums");
import CPlayGameState = require("PlayGameState");

class CMissile extends CBullet {

    MISSILE_FRAMES: number = 2;

    constructor(theScene: CScene, playGameState: CPlayGameState) {
        super(playGameState);
        this.m_name = "Missile";
    }

    //-------------------------------------------------------------

    public getActorInfo() {
        this.m_actorInfo = this.m_scene.GetlistOfActors();
        return this.m_actorInfo.GetActorInfoListItem(enums.ActorInfoType.INFO_MISSILE);
    }

    //-------------------------------------------------------------

    // Get the Speed Vector
    public getSpeed() {
        this.m_actorInfo = this.m_scene.GetlistOfActors();
        return this.m_actorInfo.GetActorInfoListItem(enums.ActorInfoType.INFO_MISSILE).m_speed.y;
    }

    //-------------------------------------------------------------

    public update(controls: gsCControls)
    {
        this.m_position.x = this.m_position.x + this.m_velocity.x;
        this.m_position.y = this.m_position.y + this.m_velocity.y;
        this.m_sprite.setFrame(this.MISSILE_FRAMES * this.m_grade);
        return true;
    }
}
export = CMissile;

File: Xenon-Original_C++_Code/xenon/source/missile.cpp
//-------------------------------------------------------------
//
// Class:	CMissile
//
// Author:	John M Phillips
//
// Started:	06/05/00
//
// Base:	CBullet
//
// Derived:	None
//
//-------------------------------------------------------------

#include "game.h"

//-------------------------------------------------------------

CMissile::CMissile()
{
}

//-------------------------------------------------------------

CMissile::~CMissile()
{
//	CBullet::~CBullet();
}

//-------------------------------------------------------------

bool CMissile::update(Controls *controls)
{
	m_position += m_velocity;

	m_sprite.setFrame(MISSILE_FRAMES * m_grade);

	return true;
}

//-------------------------------------------------------------
